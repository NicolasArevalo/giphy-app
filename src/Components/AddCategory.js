import React, { useState } from 'react';
import PropTypes from 'prop-types';
//import setCategories from '../GifExpertApp';


export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');//si se deja sin las comillas queda valiendo undefined y da error xD

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    };

    const handleSubmit = (e) =>{
        e.preventDefault(); 
        //Permite quitar el comportamiento por defecto 
        //del submit del formulario sque refresca toda la pÃ¡gina xD

        if( inputValue.trim().length > 2){
            setCategories( categories => [ inputValue, ...categories ]);
            setInputValue('');
        }
    };


    return (

            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange }
                />
                <button type='submit'>Buscar</button>
            </form>
    );
};

AddCategory.propTypes={
    setCategories: PropTypes.func.isRequired
}
