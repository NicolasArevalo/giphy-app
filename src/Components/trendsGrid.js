import React from 'react'
import { useFetchTrends } from '../hooks/useFetchTrends'
import { GifGridItem } from './GifGridItem'

export const TrendsGrid = () => {

    const { data:images , loading } = useFetchTrends();

    return (
        <>
            <h3>Trending ðŸ”¥</h3>

            { loading && <p className='animate__animated animate__bounce'>Loading</p>} {/* Ampersan(? nos deja evaluar algo solo con respuesta positiva, el ternario (?) positivca y negativa */}


            <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem
                            key={ img.id }
                            { ...img }
                        />    
                    ))

                }
            </div>
            

        </>
    )
}
