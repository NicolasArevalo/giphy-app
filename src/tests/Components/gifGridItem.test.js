import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../Components/GifGridItem';

describe('Pruebas den GifGriditem', ()=>{

    const title = 'un titulo';
    const url = 'https:localhost/algo.jpg';

    test('should show correctly the component ', () => {
        const wrapper = shallow( <GifGridItem 
            title={ title } 
            url={ url } />)
        expect( wrapper ).toMatchSnapshot();
    })
    
})