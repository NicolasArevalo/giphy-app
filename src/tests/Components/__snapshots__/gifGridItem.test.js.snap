// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas den GifGriditem should show correctly the component  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="https:localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
